contracts/libraries/ERC721Booking.sol

```diff
@@ -178,16 +245,20 @@ abstract contract ERC721Booking is Context, ERC165, IERC721, IERC721Metadata, Re
 
         _beforeTokenTransfer(from, to, fromId, toId);
 
-        // Underflow of the sender's balance is impossible because we check for
-        // ownership above and the recipient's balance can't realistically overflow.
         uint256 amount = toId - fromId + 1;
-        if (_bookedBy[fromId] != address(0)) {
+        if (_bookedBy[fromId] == address(0)) {
```

Comment:
There is duplication between code that starts in this line and code that starts in line 176.  Perhaps we should consider factoring this code out into an auxiliary internal function that is used by both the single token transfer and by the bulk transfer.